Check ruby-position when checking for emphasis mark / ruby conflicts

fantasai opened the pull request:
#### 1bd2e3ab2ace6e6d4776deaa96e0e8313c71f2dd
<pre>
Check ruby-position when checking for emphasis mark / ruby conflicts
<a href="https://bugs.webkit.org/show_bug.cgi?id=279219">https://bugs.webkit.org/show_bug.cgi?id=279219</a>
<a href="https://rdar.apple.com/problem/135366189">rdar://problem/135366189</a>

Reviewed by Alan Baradlay.

Now that we support ruby-position, we need to check the position of the
annotation, not just assume that it is on the over side.

* LayoutTests/fast/text/emphasis-avoid-ruby-001-expected.txt: Renamed from LayoutTests/fast/text/emphasis-avoid-ruby-expected.txt.
* LayoutTests/fast/text/emphasis-avoid-ruby-001.html: Renamed from LayoutTests/fast/text/emphasis-avoid-ruby.html.
* LayoutTests/fast/text/emphasis-avoid-ruby-002-expected.html: Added.
* LayoutTests/fast/text/emphasis-avoid-ruby-002.html: Added.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::emphasisMarkExistsAndIsAbove):

Canonical link: <a href="https://commits.webkit.org/283282@main">https://commits.webkit.org/283282@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec538580562d9677feffa8b439defaa90ccfe1d7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65780 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45153 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18399 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69806 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16389 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/67898 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52952 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16671 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52801 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11382 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68847 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41681 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56938 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33430 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38356 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14318 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15265 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60210 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14660 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71512 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9735 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14088 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60118 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9767 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57004 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60397 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14507 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8034 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1692 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40961 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42037 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43220 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41781 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
no WPT? :(

alanbaradlay commented on Source/WebCore/rendering/RenderText.cpp:
```diff
@@ -2131,10 +2131,9 @@ std::optional<bool> RenderText::emphasisMarkExistsAndIsAbove(const RenderText& r
     if (auto* annotation = findRubyAnnotation()) {
-        // The emphasis marks over are suppressed only if there is a ruby annotation box and it is not empty.
-        if (annotation->hasLines())
+        // The emphasis marks are suppressed only if there is a ruby annotation box on the same side and it is not empty.
+        if (annotation->hasLines() && isAbove == (annotation->style().rubyPosition() == RubyPosition::Over))
             return { };
-        return isAbove;
```
ok, so this is all about painting. makes sense (again, I am going to get rid of this function)

fantasai committed:
Check ruby-position when checking for emphasis mark / ruby conflicts
https://bugs.webkit.org/show_bug.cgi?id=279219
rdar://problem/135366189

Reviewed by Alan Baradlay.

Now that we support ruby-position, we need to check the position of the
annotation, not just assume that it is on the over side.

* LayoutTests/fast/text/emphasis-avoid-ruby-001-expected.txt: Renamed from LayoutTests/fast/text/emphasis-avoid-ruby-expected.txt.
* LayoutTests/fast/text/emphasis-avoid-ruby-001.html: Renamed from LayoutTests/fast/text/emphasis-avoid-ruby.html.
* LayoutTests/fast/text/emphasis-avoid-ruby-002-expected.html: Added.
* LayoutTests/fast/text/emphasis-avoid-ruby-002.html: Added.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::emphasisMarkExistsAndIsAbove):

Canonical link: https://commits.webkit.org/283282@main
